test/topology_custom/test_mv_delete_partitions.py

```diff
@@ -0,0 +1,59 @@
+
+from test.pylib.manager_client import ManagerClient
+
+import asyncio
+import pytest
+import time
+import logging
+
+logger = logging.getLogger(__name__)
+
+async def insert_with_concurrency(cql, key, values, concurrency):
+    logger.info(f"Starting writes for key {key} with concurrency {concurrency}")
+    async def do_inserts(i: int):
+        # Insert all rows with keys with the same modulo as i
+        while i < values:
+            await cql.run_async(f"INSERT INTO ks.tab (key, c, v) VALUES ({key}, {i}, {key + i})")
+            i += concurrency
+    tasks = [asyncio.create_task(do_inserts(i)) for i in range(concurrency)]
+
+    await asyncio.gather(*tasks)
+    logger.info(f"Finished writes for key {key} with concurrency {concurrency}")
+
+async def wait_for_view(cql, base_table, view_table):
+    logger.info(f"Waiting for view {view_table} to be built")
+    while True:
+        base_count = await cql.run_async(f"SELECT count(*) FROM {base_table} USING TIMEOUT 1h", timeout=600)
+        view_count = await cql.run_async(f"SELECT count(*) FROM {view_table} USING TIMEOUT 1h", timeout=600)
+        if base_count[0][0] == view_count[0][0]:
+            logger.info(f"View {view_table} has been built")
+            return
+        time.sleep(0.2)
+
+# This test reproduces issue #12379
```

Comment:
Maybe you can add a few sentences explaining the scenario you are doing here.
If I understand correctly:
You have a base table with 50 views. You create 50 partitions with 1000 rows, so 50*1000 base rows (and 50*50*1000 view rows). Then delete all 50 partitions at the same time, which leads to 50*50*1000 view rows deletions, which before this patch happened with too high concurrency and caused an OOM. A successful run of the test just needs to finish without crashing Scylla.

Reply:
I added a short description